pkg/apiserver/rest/webservice/authentication.go

```diff
@@ -58,11 +61,55 @@ func (c *authenticationWebService) GetWebService() *restful.WebService {
 		Returns(200, "", apis.DexConfigResponse{}).
 		Returns(400, "", bcode.Bcode{}).
 		Writes(apis.DexConfigResponse{}))
+
+	ws.Route(ws.GET("/refreshToken").To(c.refreshToken).
+		Doc("refresh token").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Returns(200, "", apis.RefreshTokenResponse{}).
+		Returns(400, "", bcode.Bcode{}).
+		Writes(apis.RefreshTokenResponse{}))
+
+	ws.Route(ws.GET("/loginType").To(c.getLoginType).
+		Doc("get login type").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Returns(200, "", apis.GetLoginTypeResponse{}).
+		Returns(400, "", bcode.Bcode{}).
+		Writes(apis.GetLoginTypeResponse{}))
 	return ws
 }
 
+func authCheckFilter(req *restful.Request, res *restful.Response, chain *restful.FilterChain) {
+	tokenHeader := req.HeaderParameter("Authorization")
+	if tokenHeader == "" {
+		bcode.ReturnError(req, res, bcode.ErrNotAuthorized)
+		return
+	}
+	splitted := strings.Split(tokenHeader, " ")
+	if len(splitted) != 2 {
+		bcode.ReturnError(req, res, bcode.ErrNotAuthorized)
+		return
+	}
+
+	token, err := usecase.ParseToken(splitted[1])
+	if err != nil {
+		bcode.ReturnError(req, res, err)
+		return
+	}
+	if token.GrantType != usecase.GrantTypeAccess {
+		bcode.ReturnError(req, res, bcode.ErrNotAccessToken)
+		return
+	}
+
```

Comment:
Lack of token validity check? There are get the user info by token and set the user info to context values.